snap/squashfs/squashfs.go

```diff
@@ -273,6 +275,131 @@ func (s *Snap) ReadFile(filePath string) (content []byte, err error) {
 	return content, nil
 }
 
+var sdVer int
+
+func init() {
+	// this also avoids calling systemctl all the time which perturbs
+	// some unrelated tests
+	v, err := systemd.Version()
+	if err != nil {
+		v = 0
+	}
+	sdVer = v
+}
+
+var (
+	sysGeteuid = sys.Geteuid
+)
+
+func sandboxParams(sdVer int) (params []string) {
+	if sysGeteuid() != 0 {
+		return nil
+	}
+	// effetive user is 0, use systemd-run to sandbox unsquashfs
```

Comment:
A very silly question - parts of the need for strong protection comes from the fact that this runs as root. If we would switch this to run with `--uid=<uid-of-daemon-user>`  (daemon is uid=1 on debian/ubuntu is "daemon", and  uid=2  on fedora) we might be able to simplify this this way? According to https://refspecs.linuxbase.org/LSB_3.1.1/LSB-Core-generic/LSB-Core-generic/usernames.html this is a required user and even if we don't do anything else (beside the private /tmp) we would get a lot of extra security? wdyt? this could also allow us to disable some of the advanced feature that interfere badly with lxd/containers?

Reply:
as we discussed offline we write snaps as root:root